pkg/ipcache/kvstore.go

```diff
@@ -428,17 +455,27 @@ func ipIdentityWatcher(listeners []IPIdentityMappingListener) {
 					listener.OnIPIdentityCacheChange(cacheModification, oldIPIDPair, ipIDPair)
 				}
 			}
-		}
 
-		log.Debugf("%s closed, restarting watch", watcher.String())
+		case <-iw.stop:
+			// identity watcher was stopped
+			watcher.Stop()
```

Comment:
`watcher.Stop()` isn't fully thread safe but I'm pretty sure we only call `.Stop` on this instance from this code so we're using it single-threaded. Still, if you wanted to improve that behaviour a context might do all the bookkeeping for you, or just do what I suggested for `IPIdentityWatcher.Close`

Reply:
I've added a `sync.Once` to make it thread safe although it should never be used as such.